app/services/articles/feed.rb

```diff
@@ -0,0 +1,43 @@
+module Articles
+  class Feed
+    attr_reader :stories
+
+    def initialize(number_of_articles:, page:, tag: nil)
+      @number_of_articles = number_of_articles
+      @page = page
+      @tag = tag
+      @stories = published_articles_by_tag
+    end
+
+    def published_articles_by_tag
+      articles = Article.published.limited_column_select.page(@page).per(@num_articles)
+      articles = articles.cached_tagged_with(@tag) if @tag.present? # More efficient than tagged_with
+      articles
+    end
+
+    def time_based_feed(timeframe:)
```

Comment:
it'd be great to comment what timeframe is supposed to be... a string? a datetime range? it could also have a default

Reply:
The values come from `TimeFramer`, will comment

Reply:
I feel like `time_based_feed` could imply chronological? Perhaps something like `top_for_time_period_feed` or something along those lines?